test/plugins/test-trace-hapi.js

```diff
@@ -308,6 +308,41 @@ describe('hapi', function() {
           });
         });
       });
+
+      it('should end spans when client aborts request', function(done) {
```

Comment:
This is not adding up for me. In the hapi plugin we are adding events on the raw request object (aka. http.IncomingMessage). If this request object is not getting the `end` event fired for hapi, then I am not sure how restify or express could work.

One possible scenario is that for restify and express the 'req' object passed to our middleware isn't the actual raw request object but some other that is guaranteed to receive the `end` event. This doesn't seem plausible to me.